File: android/src/main/java/io/github/flutteropencc/flutteropencc/FlutterOpenccPlugin.java
package io.github.flutteropencc.flutteropencc;

import android.app.Activity;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

import com.zqc.opencc.android.lib.ChineseConverter;
import com.zqc.opencc.android.lib.ConversionType;

/**
 * FlutterOpenccPlugin
 */
public class FlutterOpenccPlugin implements MethodCallHandler {
    private final Activity activity;

    private FlutterOpenccPlugin(Activity activity) {
        this.activity = activity;
    }

    /**
     * Plugin registration.
     */
    public static void registerWith(Registrar registrar) {
        final MethodChannel channel = new MethodChannel(registrar.messenger(), "flutter_opencc");
        channel.setMethodCallHandler(new FlutterOpenccPlugin(registrar.activity()));
    }

    @Override
    public void onMethodCall(MethodCall call, Result result) {
        if (call.method.equals("convert")) {
            ConversionType conversionType = ConversionType.S2T;
            String originalText = call.argument("text");
            String config = call.argument("config");
            switch (config) {
                case "t2s":
                    conversionType = ConversionType.T2S;
                    break;
                case "s2tw":
                    conversionType = ConversionType.S2TW;
                case "tw2s":
                    conversionType = ConversionType.TW2S;
                    break;
                case "s2hk":
                    conversionType = ConversionType.S2HK;
                    break;
                case "s2twp":
                    conversionType = ConversionType.S2TWP;
                    break;
                case "tw2sp":
                    conversionType = ConversionType.TW2SP;
                    break;
                case "t2hk":
                    conversionType = ConversionType.T2HK;
                    break;
                case "t2tw":
                    conversionType = ConversionType.T2TW;
                    break;
                default:
                    break;
            }
            String res = ChineseConverter.convert(originalText, conversionType, activity.getApplicationContext());
            result.success(res);
        } else {
            result.notImplemented();
        }
    }
}

File: ios/Classes/FlutterOpenccPlugin.m
#import "FlutterOpenccPlugin.h"
#import "OpenCCService.h"
@implementation FlutterOpenccPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterMethodChannel* channel = [FlutterMethodChannel
                                     methodChannelWithName:@"flutter_opencc"
                                     binaryMessenger:[registrar messenger]];
    FlutterOpenccPlugin* instance = [[FlutterOpenccPlugin alloc] init];
    [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
    NSString *str = call.arguments[@"text"];
    NSString *config = call.arguments[@"config"];
    OpenCCServiceConverterType converterType = OpenCCServiceConverterTypeS2T;
    
    if (config != nil) {
        NSArray *configs = @[@"s2t",@"t2s",@"s2tw",@"tw2s",@"s2hk",@"hk2s",@"s2twp",@"tw2sp",@"t2hk",@"t2tw"];
        NSUInteger item = [configs indexOfObject: config];
        switch (item) {
            case 0:
                converterType =OpenCCServiceConverterTypeS2T;
                break;
            case 1:
                converterType = OpenCCServiceConverterTypeT2S;
                break;
            case 2:
                converterType = OpenCCServiceConverterTypeS2TW;
                break;
            case 3:
                converterType = OpenCCServiceConverterTypeTW2S;
                break;
            case 4:
                converterType = OpenCCServiceConverterTypeS2HK;
                break;
            case 5:
                converterType = OpenCCServiceConverterTypeHK2S;
                break;
            case 6:
                converterType = OpenCCServiceConverterTypeS2TWP;
                break;
            case 7:
                converterType = OpenCCServiceConverterTypeTW2SP;
                break;
            case 8:
                converterType = OpenCCServiceConverterTypeT2HK;
                break;
            case 9:
                converterType = OpenCCServiceConverterTypeT2TW;
                break;
            default:
                converterType = OpenCCServiceConverterTypeS2T;
                break;
        }
        
    }
    OpenCCService *openCCService = [[OpenCCService alloc]initWithConverterType:converterType];
    
    if ([@"convert" isEqualToString:call.method]) {
        result([openCCService convert: str]);
    } else {
        result(FlutterMethodNotImplemented);
    }
}

@end
